db/view/view.cc

```diff
@@ -518,18 +518,37 @@ row_marker view_updates::compute_row_marker(const clustering_or_static_row& base
      *      will all unselected columns.
      */
 
-    // WARNING: The code assumes that if multiple regular base columns are present in the view key,
-    // they share liveness information. It's true especially in the only case currently allowed by CQL,
-    // which assumes there's up to one non-pk column in the view key. It's also true in alternator,
-    // which does not carry TTL information.
+    // We need to pick one liveness (timestamp and expiration) for the row
```

Comment:
I guess comment above is not accurate (never was). Esp. case 1) . In general I find case 2 and 3 not helpful in understanding what's going on here :). But since it was already there I don't insist on changing it.

Reply:
I looked at the comment above. It is not wrong, it just describes the usual CQL case, where there is zero or one new key column in the view (i.e., key column in the view which was a regular column in the base). Item 1 is about the "one" case, and item 2 and three are about the "zero" case.

The separate comment, which I modified, was about the "two" case. The "two" case is also the topic of this patch, which is why the comment on "one" and "zero" didn't fit the discussion we had about two.

I'll try to think how to combine all both these comments into one coherent comment.

Reply:
On second thought, item 3 is wrong. It was completely broken, and replaced long ago (commit 6c0034138383f4e596e83a661c22fe255fc395b5 got rid of the code this comment refers to) by "virtual columns".

I'll rewrite the entire comment.

Reply:
Not that I can offer any solution :/ but looks like in this case it is due to 'weakness' of the code, it's not self explanatory and tries to silently (apart from the comment) handle different sets of constraints.

Reply:
I'll send a new version of the patch and the comment, but I'm no longer convinced that the whole implementation is 100% correct. I think maybe the whole two-column schtick doesn't really work correctly with repair. I wrote my doubts in https://github.com/scylladb/scylladb/issues/17190. But in any case, my new implementation in this PR should work if repairs don't happen, and it's much better than today's implementation which was completely broken even without repairs, as a user discovered.